cpp.md

```diff
@@ -0,0 +1,163 @@
+
+We are also using CPP check to avoid any false positives in the project. You can easily install it on your system and learn more about it from - http://cppcheck.sourceforge.net/
+
+### Coding Style
+
+**Class name -** use camel case with first upper. eg
+``` C++
+class Person
+{
+  // ..
+};
+```
+
+**local variable name -**  use snake type. eg
+``` C++
+std::string first_name = “sameer”
+```
+
+**local class variables -** should be camel case with first letter lower case, eg
+``` C++
+SampleClass sampleObj;
+```
+
+**member class variables -** these are member class variables (objects), should be camel case with first letter lower case  and prefixed with m_.  public class variables should be named as local class variables
+``` C++
+SampleClass *m_sampleObj;
+```
+
+**function names -** use camelcase with first letter lower, eg:
+``` C++
+void doSomethingForMe ()
+{
+  // ..
+}
+```
+
+**function params -** optionally, you can precede params with p_ when params in function scope named similar to params to avoid confusion. params follow same naming convention as local variables. 
+``` C++
+void doSomethingForMe (float p_img_height)
+{
+  img_height = static_cast<int>(p_img_height);
+}
+```
+
+**namespaces -** should be camel case:
+``` C++
+namespace CoreConfig 
+{
+    // ... 
+}
+```
+
+**constants -** identifiers should be all caps. eg:
+``` C++
+const int MIN_AGE = 18;
+```
+
+**File Name -** In some places, file can contain multiple class with a main class and small supporting classes, unless these classed are being used by other classes as well, you can keep them in same file to prevent cluttering in file system. The name of the file will be same as the main class in same camel case.
+
+**Includes -** CPP includes generally have .hpp extension while C uses .h extension. However, in our case, we will be using .h.
+please keep all the includes in .h files only! reason for that is to avoid duplicate include statements, though there will be include guards, that doesn’t look nice, and someone does not have to follow 2 different files to check includes. Also, .h files is where we declare class variables, so having includes there will help you.
+ALL .h FILES MUST HAVE INCLUDE GUARDS.
+
+**scope -** start scope {} with new line. eg:
+``` C++
+namespace SampleNS
+{
+  class SampleClass : public OtherClass, public AnotherClass
+  {
+    //..
+  };
+}
+```
+
+for branching and looping, if you have just one statement, and cannot fit in one line, please use scope for it to avoid confusions. eg:
+``` C++
+for (int i = 0; i < 10; i++)
+  if (i % 2 == 0)
+    std:;cout << i << std::endl;
+```
+this will work, however, one must look closely to understand it. Instead this is easy on eyes:
+``` C++
+for (int i = 0; i < 10; i++)
+{
+  if (i % 2 == 0)
+  {
+    std:;cout << i << std::endl;
+  }
+}
+```
+however, you can still use inline branching, for eg:
+``` C++
+if (this->user.getAge() < Globals::MIN_AGE) return false;
```

Comment:
is this okay without braces ? looks difficult to read

Reply:
as long as it is in same line, it is fine. many standards like google say the same, also I read a medium article somewhere which said the same thing. personally, i also find this fine to read.

Reply:
cool. Please share the medium article if you can

Reply:
https://medium.com/@jonathanabrams/single-line-if-statements-2565c62ff492